src/components/Dashboard/DashboardDevice.tsx

```diff
@@ -0,0 +1,231 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import React, { useMemo } from 'react';
+import { CompositeFeature } from 'types';
+import { BaseFeatureProps } from 'components/features/base';
+import DeviceFooter from 'components/Dashboard/DeviceFooter';
+
+import styles from 'components/Dashboard/DashboardDevice.scss';
+import ReactSwitch from 'react-switch';
+
+type Props = BaseFeatureProps<CompositeFeature>;
+
+type State = {
+    isThermostat?: true;
+    isSocket?: true;
+    hasTemperature?: true;
+    hasHumidity?: true;
+    hasContact?: true;
+    hasSwitch?: true;
+    hasOccupancy?: true;
+    hasWaterLeak?: true;
+    hasClick?: true;
+};
+
+const DashboardDevice: React.FC<Props> = (props) => {
+    const state = useMemo(() => {
+        const state: Partial<State> = {};
+        if (Array.isArray(props.feature.features)) {
+            props.feature.features.forEach((config) => {
+                if (config.type === 'climate') {
+                    state.isThermostat = true;
+                } else if (config.property === 'power' || config.property === 'energy') {
+                    state.isSocket = true;
+                }
+                switch (config.property) {
```

Comment:
you should use `config.name`, since property name can contain endpoint name

Reply:
why not simplify all this and use config.name? without computing all this boolean properties?
here you can see all possible names
https://github.com/Koenkk/zigbee-herdsman-converters/blob/4ce85ba330a274c9943164259b14c5a56dd2e0f3/lib/exposes.js#L424

Reply:
nice! yeah as I mention my code was more than a year ago :D so mostly copy-pasted my current implementation to show & tell so I will not reimplement the wheel! will check and update

Reply:
> you should use `config.name`, since property name can contain endpoint name
```
> const c = require("zigbee-herdsman-converters")
undefined
> c.exposes.numeric("temperature")
Numeric {
  type: 'numeric',
  name: 'temperature',
  property: 'temperature',
  access: undefined
}
> c.exposes.numeric("temperature").withEndpoint("sens1")
Numeric {
  type: 'numeric',
  name: 'temperature',
  property: 'temperature_sens1',
  access: undefined,
  endpoint: 'sens1'
}```